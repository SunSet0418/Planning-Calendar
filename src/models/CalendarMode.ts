export enum CalendarMode {
    Month = '월',
    Week = '주'
}