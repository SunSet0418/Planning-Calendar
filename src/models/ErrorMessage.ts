export class ErrorMessage {
    constructor(
        public title: string,
        public description: string
    ) { }
}