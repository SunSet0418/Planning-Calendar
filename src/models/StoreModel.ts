import { TaskStoreModel } from '../modules/reducers/task';

export interface StoreModel {
    task: TaskStoreModel
}