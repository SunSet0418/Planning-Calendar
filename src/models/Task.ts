export interface Task {
    id: number
    startDate: string,
    endDate: string,
    startTime: number,
    durationTime: number
}