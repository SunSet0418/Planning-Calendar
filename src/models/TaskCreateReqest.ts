export interface TaskCreateRequest {
    name: string,
    date: string,
    startHour: number,
    endHour: number
}