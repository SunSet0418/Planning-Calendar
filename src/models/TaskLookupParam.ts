export class TaskLookupParam {
    constructor(
        public year: number,
        public month: number
    ) {}
}