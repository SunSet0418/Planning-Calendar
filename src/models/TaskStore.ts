import {Task, ErrorMessage} from '.';
export class TaskStore {
    constructor(
        public tasks: Task[] = [],
        public isTaskLoading: boolean = false,
        public isTaskLoaded: boolean = false,
        public isTaskUpdating: boolean = false,
        public isTaskUpdated: boolean = false,
        public error?: ErrorMessage
    ) { }
}