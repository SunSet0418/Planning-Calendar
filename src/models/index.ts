export { TaskStore } from './TaskStore';
export { Task } from './Task';
export { StoreModel } from './StoreModel';
export { CalendarMode } from './CalendarMode';
export { ErrorMessage } from './ErrorMessage';