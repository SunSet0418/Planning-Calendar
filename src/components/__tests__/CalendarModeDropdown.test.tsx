import * as React from 'react';
import * as Enzyme from 'enzyme';
import { CalendarMode } from '../../models/CalendarMode';
import { CalendarModeDropdown } from '../CalendarModeDropdown';

describe('<CalendarModeDropdown />', () => {
    let mode = CalendarMode.Month;
    const changeModeMock = jest.fn((newMode: CalendarMode) => {mode = newMode});
    let component = null;

    it('renders correctly', () => {
        component = Enzyme.shallow(<CalendarModeDropdown currentMode={mode} onChangeMode={changeModeMock} />);
    });

    it('should match snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('should 월', () => {
        expect(component.find('span').at(0).text()).toBe('월');
    });
});