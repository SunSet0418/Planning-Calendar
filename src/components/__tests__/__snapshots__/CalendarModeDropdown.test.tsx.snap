// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarModeDropdown /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    currentMode="월"
    onChangeMode={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        style={
                Object {
                        "marginLeft": 8,
                      }
        }
>
        <span>
                월
        </span>
         
        <Icon
                type="down"
        />
</Button>,
      "mouseEnterDelay": 0.15,
      "mouseLeaveDelay": 0.1,
      "overlay": <Menu
        onClick={[Function]}
>
        <MenuItem>
                월
        </MenuItem>
        <MenuItem>
                주
        </MenuItem>
</Menu>,
      "placement": "bottomLeft",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "block": false,
        "children": Array [
          <span>
            월
</span>,
          " ",
          <Icon
            type="down"
/>,
        ],
        "ghost": false,
        "htmlType": "button",
        "loading": false,
        "style": Object {
          "marginLeft": 8,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "월",
          },
          "ref": null,
          "rendered": "월",
          "type": "span",
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "type": "down",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          style={
                    Object {
                              "marginLeft": 8,
                            }
          }
>
          <span>
                    월
          </span>
           
          <Icon
                    type="down"
          />
</Button>,
        "mouseEnterDelay": 0.15,
        "mouseLeaveDelay": 0.1,
        "overlay": <Menu
          onClick={[Function]}
>
          <MenuItem>
                    월
          </MenuItem>
          <MenuItem>
                    주
          </MenuItem>
</Menu>,
        "placement": "bottomLeft",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": false,
          "children": Array [
            <span>
              월
</span>,
            " ",
            <Icon
              type="down"
/>,
          ],
          "ghost": false,
          "htmlType": "button",
          "loading": false,
          "style": Object {
            "marginLeft": 8,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "월",
            },
            "ref": null,
            "rendered": "월",
            "type": "span",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "type": "down",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
